### DISCUSSION TITLE
Analytics: Undefined symbols for architecture x86_64

### DISCUSSION THREAD

**Discussion by hayata-suenaga** [invertase/react-native-firebase#6484] [Q&A] [answered]
[Stackoverflow post](https://stackoverflow.com/questions/73459599/react-native-firebase-analytics-undefined-symbols-for-architecture-x86-64)

I'm trying to build an iOS development build of my Expo project for an iOS simulator. I'm developing the app using the managed workflow with a custom development client.

My project depends on a suite of [React Native Firebase libraries](https://rnfirebase.io/), including Firestore and Auth. Recently, I also added [Analytics](https://rnfirebase.io/analytics/usage) by executing `expo install @react-native-firebase/analytics`. When I tried to build the project with `expo run:ios`, I got the following error.

```
❌  Undefined symbols for architecture x86_64
┌─ Symbol: _OBJC_CLASS_$_FIRAnalytics
└─ Referenced from: objc-class-ref in RNFBAnalyticsModule.o


❌  ld: symbol(s) not found for architecture x86_64



❌  clang: error: linker command failed with exit code 1 (use -v to see invocation)


    Capabilities for Signing & Capabilities may not function correctly because its entitlements use a placeholder team ID. To resolve this, select a development team in the my_project_name editor. (in target 'my_project_name' from project 'my_project_name')
```

[This is the link to the minimum, reproducible example](https://github.com/hayata-suenaga/analytics) on github. Please run `expo run:ios` after `expo install`.

 - Tried to clear caches by following instructions on [expo doc](https://docs.expo.dev/troubleshooting/clear-cache-macos-linux/#expo-cli-and-npm).
 - Deleted the `ios` directory.
 - Manually installed pods by running `pod install` inside the `ios` directory.

None of the above solutions worked. Am I missing something here?

**Comment by mikehardy** [comment]
You missed an mvce https://stackoverflow.com/help/mcve if you posted on stackoverflow, and it would help here. Some sequence of commands we could run to reproduce.

Expo support here is community driven though, and I'm not sure anyone that has contributed it so far (in the form of config plugins) is that active here. I'm the main maintainer and I have no Expo experience and no time to gain any so I'm not much help unfortunately. How did it go when you asked on the Expo forums?

I will say that just as a raw compilation error, it appears maybe the compiler is uncertain about which architectures to build and built either x86 or arm or something for FIRAnalytics. That's vague and probably wrong, sorry. This is one of those errors where Expo has helpfully isolated you from all the underlying machinery in use until something 3 levels down blows up and now there's little ability to understand it or effect change and fix it...

**Reply by hayata-suenaga** [reply]
@mikehardy Thank you for reminding me to make a mvce :) I created one. I'll also try Expo forums, too.

**Reply by mikehardy** [reply]
For what it's worth (not much, since you are on expo), here's the MVCE I use for all build testing for react-native-firebase (and incidentally to chase issues upstream into main react-native repo) - anything like this would work: https://github.com/mikehardy/rnfbdemo/blob/main/make-demo.sh

**Reply by hayata-suenaga** [reply]
ah I didn't know using bash script for MVCE is the convention.

**Reply by mikehardy** [reply]
It's not the convention, it's just one way to do it, I have to do MCVEs / build checks / failure reproductions literally multiple times a day to probe build failures like yours for all the users here so I've scripted it

**Comment by hayata-suenaga** [comment, marked as answer]
It turned out that my setting for a [config plugin](https://docs.expo.dev/guides/config-plugins/) (that writes configurations for native projects) was wrong.

`firebase-ios-sdk` requires [the usage of frameworks](https://rnfirebase.io/#altering-cocoapods-to-use-frameworks). To specify this on a managed Expo project, you need to install a config plugin called `expo-build-properties` and edit the plugin property of `app.json` like following:

```
[
  "expo-build-properties",
  {
    "ios": {
      "useFrameworks": "static"
    }
  }
]
```

In my original code, instead of `static`, I specified `dynamic`, which was causing the error.

**Reply by mikehardy** [reply]
I only recently learned myself that linkage should be static, but I was under the impression it was only important for hermes: https://github.com/reactwg/react-native-releases/discussions/26#discussioncomment-3441178

Either way, we've updated the docs here to mention static via another person helpfully contributing a PR #6488 so hopefully that helps others
